app/src/main/java/com/example/criminalintent/presentation/crime_detail/CrimeFragment.kt

```diff
@@ -203,7 +249,12 @@ class CrimeFragment : Fragment(), FragmentResultListener {
                 .setMessage(R.string.dialog_message)
                 .setNegativeButton(
                     R.string.dialog_negative
-                ) { _, _ -> activity?.supportFragmentManager?.popBackStack() }
+                ) { _, _ ->
+                    activity?.supportFragmentManager?.popBackStack()
+                    if (viewModel.crimeLiveData.value?.photoFileName == null) {
```

Comment:
This logic should not be tied to the dialog. A better place for it is in `onDestroy`.
The code to delete the file should be in viewModel. You can create a method called `releaseResources` or something like that in the view model